py-polars/polars/_utils/construction/dataframe.py

```diff
@@ -1163,14 +1164,19 @@ def arrow_to_pydf(
     column_names, schema_overrides = _unpack_schema(
         (schema or data.schema.names), schema_overrides=schema_overrides
     )
-
     try:
         if column_names != data.schema.names:
             data = data.rename_columns(column_names)
     except pa.lib.ArrowInvalid as e:
         msg = "dimensions of columns arg must match data dimensions"
         raise ValueError(msg) from e
 
+    # arrow tables allow duplicate names; we don't
+    if len(column_names) != len(set(column_names)):
```

Comment:
Can we check the width of the Polars DataFrame with the width of the Pyarrow table? That would save allocating the hashset, I think.

Reply:
No, because we haven't created the frame on our side yet ;)

Reply:
Done; made the check optimistic (after frame creation) so we can use `.width`.